packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,445 @@
+
+    // if there is a roll Up pill, set tabindex to -1
+    if (rollUpPill) {
+      PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+    }
+
+    if (pills.length > 0 && focusNode.current < pills.length) {
+      PillsUtils.setPillsTabIndex(pills, '-1');
+      currentPill.current = pills[focusNode.current].id;
+      setTabIndex('0');
+    } else if (isCollapsible && rollUpPill) { // if the first pill is rollUp pill, set rollUp pill tabindex 0
+      currentPill.current = rollUpPill.getAttribute('id');
+      setTabIndex('0');
+    }
+  }, [isCollapsible]);
+
+  const handleResize = useCallback((entries) => {
+    if (!Array.isArray(entries)) {
+      return;
+    }
+
+    setUpdatedCount(React.Children.count(children));
+    setRollUpCount(React.Children.count(children));
+
+    if (isCollapsible) {
+      generateRollUp();
+    }
+
+    handleTabIndex();
+  }, [children, isCollapsible, handleTabIndex, generateRollUp]);
+
+  useLayoutEffect(() => {
+    let observer = new ResizeObserver((entries) => {
+      handleResize(entries);
+    });
+    observer.observe(filterPillsRef.current.parentNode);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [children, handleResize, isCollapsed]);
+
+  const focusPillsContainer = () => {
+    setContainerTabindex('0');
+    filterPillsRef.current.focus();
+  };
+
+  // When a pill is deleted, focuses the new pill.
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const startIndex = PillsUtils.handleRollUp(filterPillsRef);
+
+    if (isPillDeleted.current) {
+      if (React.Children.count(children) <= 0) {
+        focusPillsContainer();
+        return;
+      }
+      // if there is a roll Up pill, set tabindex to 0
+      if (rollUpPill) {
+        PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+      }
+
+      if (pills.length > 0) {
+        PillsUtils.setPillsTabIndex(pills, '-1');
+      }
+
+      if (pills.length === startIndex) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill && startIndex >= 1) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+      isPillDeleted.current = false;
+    }
+  }, [isPillDeleted, children]);
+
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    // To focus the immediate focusable pill after the rollUp pill is selected/removed
+    if (isCollapsible && !isCollapsed) {
+      if (pills.length === React.Children.count(children)) {
+        setTabIndex('-1');
+        focusNode.current += 1;
+        currentPill.current = pills[focusNode.current].id;
```

Comment:
In the [colllapsible-filter-pill test](https://terra-framew-terra-filt-zwahkn.herokuapp.com/tests/cerner-terra-framework-docs/pills/filter-pills/collapsible-filter-pills) if I uncollapsed all the pills and close any of the two last pills that were rolled up, I get an error saying that it can't find the pill using the current focus node when trying to access the id.

![pills-error](https://user-images.githubusercontent.com/22618655/144497741-3d485a1b-7ad7-4aec-a4c9-48234cb9f2b8.gif)

Reply:
Fixed in this commit e79be980bb3defef6e9370f26558523eead90bfa